acceptancetests/deploy_stack.py

```diff
@@ -1070,7 +1070,11 @@ def dump_all_logs(self, patch_dir=None):
                     runtime_config = None
                 artifacts_dir = os.path.join(self.log_dir,
                                              client.env.environment)
-                os.makedirs(artifacts_dir)
+                if not os.path.exists(artifacts_dir):
+                    try:
+                        os.makedirs(artifacts_dir)
+                    except OSError:
```

Comment:
This exception catch is too vague. You need to make sure the OSError is due to the directory already existing (not pemissions or space issues etc).
To do this import errno and compare the exception errno to make sure it's EEXIST (17). i.e.
```python
import errno

try:
    os.makedirs(...)
except OSError as e:
     if e.errno != errno.EEXISTS:
        raise
```